docs/threat_model.md

```diff
@@ -0,0 +1,110 @@
+# Threat Model
+
+Threat modeling is a structured approach of identifying and prioritizing potential threats to a system, and determining the value that potential mitigations would have in reducing or neutralizing those threats.
+
+Using [OWASP's threat modeling cheat sheet for guidance](https://cheatsheetseries.owasp.org/cheatsheets/Threat_Modeling_Cheat_Sheet.html).
+
+## Application Entry Points
+
+The interfaces through which potential attackers can interact with the application or supply them with data.
+The diagram below models these interfaces, the sections that follow describe them further.
+
+<img src="../images/ofo_threat_model.png" alt="Diagram of OFO threat model">
+
+### Via the application to which OpenFeature Operator (OFO) sidecars flagd
+
+OFO appends flagd as a sidecar container to any pod spec application with valid annotations. The application is then able to evaluate flags by calling flagd (typically via an sdk and flag provider). While a caller of the application could be from a trusted entity (internal infrastructure), it is more prudent to presume the agent to be untrusted.
+
+### Via the Kubernetes API server
+
+OFO listens to webhooks from the Kubernetes control plane (its API server), specifically it handles mutations of pods.
+It parses the mutated pod's annotations to determine which CRD to retrieve flagd's flag configuration from, mutating flagd's internal state if necessary.
+
+## Assets
+
+### Flag source configuration custom resource (CR)
+
+Stored in the kubernetes cluster's etcd instance, this custom resource contains the configuration for the sidecar container (`flagd`) `OFO` appends to the application pod(s).
+
+### Flag configuration CR
+
+Stored in the kubernetes cluster's etcd instance, this custom resource contains the sidecar's (`flagd`) feature flags configuration. It also incorporates configuration for the sidecar container (`flagd`), although, this is deprecated and superseded by the flag source configuration CR.
+
+### Environment variables
+
+Stored in `OFO`'s container's registry, these variables are used to configure the flag source if not declared explicitly by the flag source configuration CR or flag configuration CR.
+
+### flagd flag configuration
+
+`flagd`'s flag configuration is the determinant of the response to any flag request, it is stored in `flagd`'s memory.
+There are numerous ways to set and maintain this state, configured by the flag configuration CR:
+- Within the flag configuration CR 
+- As a file
+- From a remote source
+
+## Controls
+
+### External agency
+
+#### Cluster state mutation
+
+```mermaid
+flowchart LR
+    user-agent --> kube-apiserver
+```
+
+The user-agent is typically (and therefore assumed to be) internal to the intranet. They make requests to the `kubeapi-server` to mutate the state of the cluster.
```

Comment:
I would add that in this case, the `user-agent` is a cluster admin or developer. `User-agent` is a good term to use, but we should also put it in simpler terms.

Reply:
Good shout, done